File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        mostUsedKey = Counter(nums).most_common(1)[0][0]
        return mostUsedKey
        
'''        
        hMap = {}

        for num in nums:
          if num in hMap:
            hMap[num] += 1
          else:
            hMap[num] = 1
          if hMap[num] > len(nums) // 2:
              return num
'''

File: 169-majority-element/majority-element.dart
class Solution {
  int majorityElement(List<int> nums) {
      var map = {};
      int res = 0;

      for(int n in nums){
          map[n] = 1 + (map[n] ?? 0);

            if(map[n] > nums.length/2){
                res = n;
                break;
            }

      }
    
    return res;
  }
}